internal/discovery/foreign-cluster-operator/auth.go

```diff
@@ -107,6 +107,28 @@ func (r *ForeignClusterReconciler) getRemoteClient(
 	return nil, nil
 }
 
+func (r *ForeignClusterReconciler) getRemoteTenantNamespace(fc *discoveryv1alpha1.ForeignCluster) (bool, error) {
```

Comment:
Minor comment: I was a bit confused by the name of the method. Actually, this method is not getting and returning the remoteTenantNamespace here. On the contrary, it is setting it in the foreignCluster. Maybe rename it with something like tryFetchRemoteTenantNamespace?

Reply:
Agree that the current name appears to be misleading (AFAIU you are checking if this piece of information is updated, rather than getting something). Would also add a comment to the function (although not strictly required by the linter)